website/versioned_docs/version-v1.3_alpha/troubleshooting.md

```diff
@@ -49,3 +49,113 @@ You can generate the missing `.dat` files using the following command:
 ```bash
 pnpm build:circuits-c
 ```
+
+## contracts: `prove` command failure
+
+### Case `Commitment mismatch`
+
+If your log looks like following, that's because you have already run the `prove` command before. You could either do go to `cli`, and run `genProof` command.
+
+```
+Error: commitment mismatch
+    at Prover.validateCommitment (/home/user/Documents/maci/contracts/tasks/helpers/Prover.ts:458:13)
+    at Prover.proveTally (/home/user/Documents/maci/contracts/tasks/helpers/Prover.ts:390:12)
+    at processTicksAndRejections (node:internal/process/task_queues:95:5)
+    at async SimpleTaskDefinition.action (/home/user/Documents/maci/contracts/tasks/runner/prove.ts:235:7)
+    at async Environment._runTaskDefinition (/home/user/Documents/maci/node_modules/.pnpm/hardhat@2.22.2_ts-node@10.9.2_typescript@5.4.3/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
+    at async Environment.run (/home/user/Documents/maci/node_modules/.pnpm/hardhat@2.22.2_ts-node@10.9.2_typescript@5.4.3/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
+    at async main (/home/user/Documents/maci/node_modules/.pnpm/hardhat@2.22.2_ts-node@10.9.2_typescript@5.4.3/node_modules/hardhat/src/internal/cli/cli.ts:323:7)
+ ELIFECYCLE  Command failed with exit code 1.
+ ELIFECYCLE  Command failed with exit code 1.
+```
+
+That's because the commitments differ after each `genProofs` command, which is generated base on random salt.
+
+In [core/Poll.ts](/docs/typedoc/core/classes/Poll):
+
+```
+let newSbSalt = genRandomSalt();
+while (this.sbSalts[this.currentMessageBatchIndex!] === newSbSalt) {
+  newSbSalt = genRandomSalt();
+}
+```
+
+## General Failure
+
+### Case `AssertionError`
+
+This could happen when you run `prove` in the `contracts` package, or run `genProofs` in the `cli` package. If your log looks like following, there are two possible reasons:
+
+1. Your MACI keypair for coordinator is generated wrongly, please generate a new pair and run the whole process again.
```

Comment:
Maybe mentioned that there was a breaking change in a third party package (zk-kit/eddsa-poseidon)

Reply:
@kittybest would recommend adding this, cause otherwise it can look to the users that the key generation process might be flawed

Reply:
Just added, since these two conversations are hidden so I failed to see them previously.